src/main/java/login/UserManager.java

```diff
@@ -1,41 +1,129 @@
 package login;
 
+import entities.Profile;
 import entities.User;
+import entities.UserList;
+
 
-import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
 
 public class UserManager {
-    private List<User> AllUser;
+    private UserList AllUser;
+
+    final login.LoginOutputBound LoginOutputBound;
 
-    public UserManager() {
-        AllUser = new ArrayList<>();
+    public UserManager(login.LoginOutputBound loginOutputBound) {
+        this.LoginOutputBound = loginOutputBound;
     }
 
-    public void setAllUser(String username, String password) {
-        for(User person :AllUser){
-            if (Objects.equals(person.getUsername(), username)){
-                continue;
-            } else{
-                User UserName = new User(username, password);
-                AllUser.add(UserName);
-            }
-        }
+
+    //create Account
+    public void CheckAllUser(String username,String password){
+        if (AllUser.CheckAllUser(username)){
+            AllUser.AddAllUser(username,password);
+            LoginOutputBound.CreatAccountSuccess();
+        }else{LoginOutputBound.CreatAccountFail();}
     }
 
 
-    // method to add user to the allUsers when user created the account,need to check if the username already in the list
 
     // method to find the user in the allUsers and change the status to logged in
+    public void Login(String username, String password){
+        for(User person :AllUser.getAllUser()){
+            if(Objects.equals(person.getUsername(), username) && Objects.equals(person.getPassword(), password)){
+                person.setLoginStatus(true);
+                LoginOutputBound.LoginSuccess();
+            }else{
+                LoginOutputBound.LoginFailed();
+                }
+
+            }
+        }
+
+
     // method to find the user in the allUsers and change the status to logged out
+    public void Logout(String username){
+        for(User person :AllUser.getAllUser()){
```

Comment:
For almost all the methods: login, logout, addprofile, checkfollow, follow, checkunfollow, unfollow, you are going through AllUser list to find the user. I feel like this is redundant and slow. Maybe you could have a helper method or something? Or maybe to save the logged in user as a variable in UserManager (I think this is ok since only one user can logged in at a time), so it can be used across these methods.

Reply:
I felt that even if I created a helper method, I need to call the helper method every time. Also AllUser is  already a variable in the UserManage, however it is a Userlist type (a class i created to save user data) , rather than a List type for serialization reasons.  Thank you for your advice though~